app/models/source.rb

```diff
@@ -36,18 +36,23 @@ def super_key_credential
     authentications.detect { |a| a.authtype == source_type.superkey_authtype }
   end
 
+  # reset from Application can't invoke new check in Application
+  def reset_availability!(availability_check: true)
+    reset_availability(:availability_check => availability_check)
+  end
+
```

Comment:
I'm not sure I see the reason for the `!` at the end of this method - if all it's doing is wrapping the non-! method why have it? Unless we think there is going to be more logic in this method?

Reply:
@lindgrenj6 this is overriden method `reset_availability!` from the concern, which calls `save!`, but only for Source model it's not wanted